webapp/src/js/services/telemetry.js

```diff
@@ -0,0 +1,215 @@
+var moment = require('moment'),
+  uuid = require('uuid/v4');
+
+angular
+  .module('inboxServices')
+  .factory('Telemetry', function($log, $q, $window, DB, Session) {
+    'use strict';
+    'ngInject';
+
+    var queue = $q.resolve();
+
+    // Intentionally scoped to the whole browser (for this domain). We can then
+    // tell if multiple users use the same device
+    var DEVICE_ID_KEY = 'medic-telemetry-device-id';
+
+    // Intentionally scoped to the specific user, as they may perform a
+    // different role (online vs. offline being being the most obvious) with
+    // different performance implications
+    var DB_ID_KEY = ['medic', Session.userCtx().name, 'telemetry-db'].join('-');
+    var LAST_AGGREGATED_DATE_KEY = [
+      'medic',
+      Session.userCtx().name,
+      'telemetry-date',
+    ].join('-');
+
+    var getDb = function() {
+      var dbName = $window.localStorage.getItem(DB_ID_KEY);
+      if (!dbName) {
+        dbName =
+          'medic-user-' + Session.userCtx().name + '-telemetry-' + uuid();
```

Comment:
What's the benefit of having a UUID here?  Could you just use `medic-${userName}-telemetry` as the db name directly and achieve the same?

Reply:
You know that's a great question. This may have been left over from testing when we were not deleting the old DBs. I don't remember if we still need it though, @garethbowen can you remember? It may be that `db.destroy()` has not completely cleaned up by the time you go to instantiate another DB of the same name in its place and it breaks horribly, but I think no longer using `angular-pouchdb` was all we really needed to do?

NB: about ~50% of the time Gareth and I spent pairing on this was trying to get the delete + create flow working right because it would break obscurely, so I'm a bit gun shy here ;-)

Reply:
Sounds thorny.  Consider removing the UUID if you have the patience.  The only danger in leaving it here is we may fail to clean up the databases if we lose track of a UUID.

Reply:
If it's kept, adding a comment to explain its presence would also be helpful.

Reply:
I'm fairly sure we needed it because if you destroy a PouchDB instance then create a new one with the same name it fails saying the IDB is unavailable (or something).